File: leetcode/454/454.ts
 // @ts-ignore
function fourSumCount(A: number[], B: number[], C: number[], D: number[]): number {
    var m:{[key:number]:number} = {};
    var res = 0;
    A.map(a=>B.map(b=>m[a+b] = (m[a+b] || 0) + 1))
    C.map(c=>D.map(d=>res += (m[-c-d] || 0)))
    return res
};

File: leetcode/454/454.go
package main

func fourSumCount(A []int, B []int, C []int, D []int) int {
	res := 0
	m := make(map[int]int)
	for _, a := range A {
		for _, b := range B {
			m[a+b] = m[a+b] + 1
		}
	}
	for _, c := range C {
		for _, d := range D {
			res += m[-c-d]
		}
	}
	return res
}
